### PR TITLE
Bump `black` and `ruff`

### PR DESCRIPTION
**Pull request by harshil21**
- Black has reached a new [stable release](https://black.readthedocs.io/en/stable/change_log.html#id2), which brings the new stable style. So it has lots of fixes for us :)

- Ruff has a new release sorting all our `__slots__`. 

- flake8 is angry that Black has changed so I silenced it.

- ruff now runs first in pre-commit as per @clot27's suggestion

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -75,16 +87,4 @@ repos:
...
-    -   id: ruff
-        name: ruff
-        files: ^(telegram|examples|tests)/.*\.py$
-        additional_dependencies:
-          - httpx~=0.26.0
-          - tornado~=6.4
-          - APScheduler~=3.10.4
-          - cachetools~=5.3.2
-          - aiolimiter~=1.1.0
+        - --check
```
Comment (Bibo-Joshi) [inline .pre-commit-config.yaml:90]: missing new line

File: setup.cfg [hunk 2]
```diff
@@ -12,7 +12,7 @@ upload-dir = docs/build/html
 [flake8]
 max-line-length = 99
 ignore = W503, W605
-extend-ignore = E203
+extend-ignore = E203, E704
```
Comment (Bibo-Joshi) [inline setup.cfg:15]: what kind of error is this?

Comment (harshil21) [reply setup.cfg:15]: having multiple statements in one line. got triggered cause black likes to put the Ellipsis at the same line as an overloaded function definition

**Comment by Bibo-Joshi** [comment]
The issues reported by DeepSource appear to be invalid. probably they changed the configuration again …

**Pull request merged**